packages/trpc/server/routers/viewer/slots/util.ts

```diff
@@ -70,9 +70,31 @@ export const checkIfIsAvailable = ({
 };
 
 export async function getEventType(input: TGetScheduleInputSchema) {
+  let eventTypeId = input.eventTypeId;
+  // If we only have the slug and usernameList, we need to get the id first
+  if (input.eventTypeId === undefined && input.eventTypeSlug && input.usernameList) {
+    const eventType = await prisma.eventType.findFirst({
+      where: {
+        slug: input.eventTypeSlug,
+        users: {
+          some: {
+            username: input.usernameList && input.usernameList[0],
+          },
+        },
+      },
+      select: {
+        id: true,
+      },
+    });
+    if (!eventType) {
+      throw new TRPCError({ code: "NOT_FOUND" });
+    }
+    eventTypeId = eventType.id;
+  }
```

Comment:
We should be better be fetching eventTypeId on the server than waiting in the frontend.

Reply:
Also getSchedules allows two combinations of inputs, slug and usernameList or eventTypeId. But then wasn't using  slug and usernameList combination to fetch the event.

Reply:
> We should be better be fetching eventTypeId on the server than waiting in the frontend.

Working on that 🙏